cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -575,6 +575,19 @@ impl<I: VCodeInst> MachBuffer<I> {
         }
     }
 
+    /// Similar to [`MachBuffer::register_constants`] but registers a
+    /// single constant metadata. This function is useful in
+    /// situations where not all constants are known at the time of
+    /// emission.
+    pub fn register_constant(&mut self, constant: &VCodeConstant, data: &VCodeConstantData) {
```

Comment:
now that this exists, can we rewrite `register_constants` to use it?

Reply:
Good call, fixed!